### PR TITLE
Use @api3/chains in airnode-protocol

### PR DESCRIPTION
**Pull request by bbenligiray**
Closes https://github.com/api3dao/airnode/issues/1734

I confirmed that the contracts that are already deployed pass `test:verify-local`. I will deploy and verify new contracts as a part of https://github.com/api3dao/airnode/issues/1735 , which will allow me to test further.

### CODE REVIEW

File: packages/airnode-protocol/deployments/references.json [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
-    "4": "rinkeby",
-    "5": "goerli",
+    "1": "ethereum",
+    "5": "ethereum-goerli-testnet",
     "10": "optimism",
     "30": "rsk",
     "31": "rsk-testnet",
-    "42": "kovan",
     "56": "bsc",
-    "77": "gnosis-testnet",
```
Comment (bbenligiray) [inline packages/airnode-protocol/deployments/references.json:12]: We forgot to delete this

File: packages/airnode-protocol/package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "test:gas": "REPORT_GAS=TRUE hardhat test"
   },
   "devDependencies": {
+    "@api3/chains": "https://github.com/api3dao/chains.git#new-chains",
```
Comment (bbenligiray) [inline packages/airnode-protocol/package.json:31]: We're in the process of making some big changes to @api3/chains https://github.com/api3dao/chains/pull/1/
I will update this to use the new version as soon as possible

File: packages/airnode-protocol/deployments/ethereum-sepolia-testnet/solcInputs/35d9bfaca6ee7fc01e60ce83241d99fb.json
Comment (acenolaza) [inline packages/airnode-protocol/deployments/ethereum-sepolia-testnet/solcInputs/35d9bfaca6ee7fc01e60ce83241d99fb.json:1]: This is showing `*/optimism-testnet/* -> */ethereum-sepolia-testnet/*` which I'm guessing it's some kind of weirdness of github diff due to file names matching or something...

File: packages/airnode-protocol/deployments/optimism-goerli-testnet/solcInputs/35d9bfaca6ee7fc01e60ce83241d99fb.json
Comment (acenolaza) [inline packages/airnode-protocol/deployments/optimism-goerli-testnet/solcInputs/35d9bfaca6ee7fc01e60ce83241d99fb.json:1]: Same here `*/sepolia/* -> */optimism-goerli-testnet/*`

File: packages/airnode-protocol/example.env [hunk 3]
```diff
@@ -0,0 +1,24 @@
+MNEMONIC=""
+ETHERSCAN_API_KEY_arbitrum-goerli-testnet=""
+ETHERSCAN_API_KEY_arbitrum-nova=""
+ETHERSCAN_API_KEY_arbitrum=""
+ETHERSCAN_API_KEY_avalanche-testnet=""
+ETHERSCAN_API_KEY_avalanche=""
+ETHERSCAN_API_KEY_boba-ethereum=""
```
Comment (acenolaza) [inline packages/airnode-protocol/example.env:7]: These should match the available chains in @api3/chains.CHAINS, right? We keep boba and sx deployments for reference but new deployments will only care about supported chains listed in @api3/chains

**Review by acenolaza — APPROVED** [review]
LGTM 👍🏻

File: packages/airnode-protocol/deployments/ethereum-sepolia-testnet/solcInputs/35d9bfaca6ee7fc01e60ce83241d99fb.json
Comment (bbenligiray) [reply packages/airnode-protocol/deployments/ethereum-sepolia-testnet/solcInputs/35d9bfaca6ee7fc01e60ce83241d99fb.json:1]: I think the local verification would have failed if there were any mistakes

File: packages/airnode-protocol/example.env (hunk 3 above)
Comment (bbenligiray) [reply packages/airnode-protocol/example.env:7]: I mean @api3/chains to support all chains that we do anything on https://github.com/api3dao/chains/issues/2

**Comment by bbenligiray** [comment]
Need to merge and publish https://github.com/api3dao/chains/pull/12 and bump the dependency for the CI to pass

**Comment by bbenligiray** [comment]
I'll merge this after https://github.com/api3dao/airnode/pull/1742 is merged

**Pull request merged**